### PR TITLE
Group exposed ports in ps output

### LINKED ISSUE #10302: `docker ps` in v1.4.x gives far too much output if a port range has been exposed
If I `docker run` a container with an exposed port range and then subsequently, run a `docker ps`, each and every port in that range is specified in the output.

If you have a large number of exposed ports, the output of the `docker ps` command becomes unreadable and can also take a number of seconds to print out. 

Example w/1000 ports exposed:
```
vagrant@vagrant:~$ sudo docker run -dt --expose=2000-3000 ubuntu:14.04 /bin/bash
f17ad1f26b4d3612b185c4eb5f817b42a2b03c7bc89177655fbe728ffcbaf305
vagrant@vagrant:~$ sudo docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      NAMES
f17ad1f26b4d        ubuntu:14.04        "/bin/bash"         4 seconds ago       Up 3 seconds        2768/tcp, 2009/tcp, 2922/tcp, 2966/tcp, 2485/tcp, 2283/tcp, 2328/tcp, 2687/tcp, 2045/tcp, 2808/tcp, 2582/tcp, 2708/tcp, 2906/tcp, 2942/tcp, 2780/tcp, 2186/tcp, 2348/tcp, 2384/tcp, 2065/tcp, 2469/tcp, 2021/tcp, 2518/tcp, 2693/tcp, 2914/tcp, 2950/tcp, 2590/tcp, 2158/tcp, 2194/tcp, 2297/tcp, 2459/tcp, 2495/tcp, 2392/tcp, 2055/tcp, 2099/tcp, 2011/tcp, 2812/tcp, 2856/tcp, 2930/tcp, 2578/tcp, 2138/tcp, 2075/tcp, 2479/tcp, 2031/tcp, 2586/tcp, 2748/tcp, 2902/tcp, 2946/tcp, 2784/tcp, 2182/tcp, 2308/tcp, 2380/tcp, 2061/tcp, 2025/tcp, 2429/tcp, 2594/tcp, 2697/tcp, 2910/tcp, 2954/tcp, 2558/tcp, 2118/tcp, 2190/tcp, 2293/tcp, 2419/tcp, 2491/tcp, 2396/tcp, 2998/tcp, 2051/tcp, 2015/tcp, 2247/tcp, 2320/tcp, 2364/tcp, 2001/tcp, 2526/tcp, 2122/tcp, 2166/tcp, 2203/tcp, 2724/tcp, 2760/tcp, 2405/tcp, 2562/tcp, 2607/tcp, 2643/tcp, 2106/tcp, 2142/tcp, 2304/tcp, 2340/tcp, 2546/tcp, 2425/tcp, 2461/tcp, 2029/tcp, 2986/tcp, 2502/tcp, 2663/tcp, 2223/tcp, 2700/tcp, 2744/tcp, 2788/tcp, 2019/tcp, 2796/tcp, 2114/tcp, 2150/tcp, 2217/tcp, 2091/tcp, 2415/tcp, 2451/tcp, 2598/tcp, 2613/tcp, 2657/tcp, 2958/tcp, 2994/tcp, 2253/tcp, 2510/tcp, 2554/tcp, 2752/tcp, 2312/tcp, 2368/tcp, 2409/tcp, 2683/tcp, 2728/tcp, 2926/tcp, 2481/tcp, 2041/tcp, 2287/tcp, 2962/tcp, 2974/tcp, 2039/tcp, 2237/tcp, 2273/tcp, 2336/tcp, 2435/tcp, 2471/tcp, 2938/tcp, 2732/tcp, 2776/tcp, 2570/tcp, 2130/tcp, 2372/tcp, 2633/tcp, 2677/tcp, 2102/tcp, 2146/tcp, 2300/tcp, 2344/tcp, 2388/tcp, 2421/tcp, 2465/tcp, 2069/tcp, 2740/tcp, 2982/tcp, 2667/tcp, 2227/tcp, 2506/tcp, 2542/tcp, 2704/tcp, 2499/tcp, 2059/tcp, 2792/tcp, 2110/tcp, 2154/tcp, 2095/tcp, 2411/tcp, 2455/tcp, 2514/tcp, 2550/tcp, 2617/tcp, 2653/tcp, 2918/tcp, 2198/tcp, 2213/tcp, 2257/tcp, 2990/tcp, 2756/tcp, 2316/tcp, 2351/tcp, 2711/tcp, 2896/tcp, 2267/tcp, 2627/tcp, 2829/tcp, 2716/tcp, 2356/tcp, 2819/tcp, 2891/tcp, 2174/tcp, 2839/tcp, 2534/tcp, 2263/tcp, 2623/tcp, 2869/tcp, 2859/tcp, 2352/tcp, 2712/tcp, 2895/tcp, 2881/tcp, 2089/tcp, 2449/tcp, 2845/tcp, 2934/tcp, 2978/tcp, 2071/tcp, 2439/tcp, 2035/tcp, 2821/tcp, 2865/tcp, 2811/tcp, 2855/tcp, 2899/tcp, 2445/tcp, 2085/tcp, 2849/tcp, 2831/tcp, 2875/tcp, 2825/tcp, 2861/tcp, 2815/tcp, 2851/tcp, 2801/tcp, 2079/tcp, 2233/tcp, 2277/tcp, 2332/tcp, 2376/tcp, 2431/tcp, 2475/tcp, 2970/tcp, 2772/tcp, 2574/tcp, 2134/tcp, 2637/tcp, 2673/tcp, 2736/tcp, 2809/tcp, 2170/tcp, 2879/tcp, 2530/tcp, 2805/tcp, 2871/tcp, 2178/tcp, 2538/tcp, 2835/tcp, 2373/tcp, 2632/tcp, 2676/tcp, 2733/tcp, 2777/tcp, 2236/tcp, 2272/tcp, 2337/tcp, 2470/tcp, 2939/tcp, 2975/tcp, 2038/tcp, 2571/tcp, 2131/tcp, 2434/tcp, 2935/tcp, 2979/tcp, 2070/tcp, 2438/tcp, 2034/tcp, 2519/tcp, 2692/tcp, 2915/tcp, 2951/tcp, 2591/tcp, 2159/tcp, 2195/tcp, 2296/tcp, 2458/tcp, 2494/tcp, 2393/tcp, 2054/tcp, 2098/tcp, 2010/tcp, 2529/tcp, 2608/tcp, 2680/tcp, 2925/tcp, 2961/tcp, 2169/tcp, 2248/tcp, 2284/tcp, 2482/tcp, 2042/tcp, 2841/tcp, 2885/tcp, 2581/tcp, 2783/tcp, 2149/tcp, 2905/tcp, 2941/tcp, 2185/tcp, 2387/tcp, 2509/tcp, 2228/tcp, 2066/tcp, 2022/tcp, 2668/tcp, 2595/tcp, 2696/tcp, 2911/tcp, 2955/tcp, 2559/tcp, 2119/tcp, 2191/tcp, 2292/tcp, 2418/tcp, 2490/tcp, 2397/tcp, 2999/tcp, 2050/tcp, 2014/tcp, 2569/tcp, 2648/tcp, 2684/tcp, 2921/tcp, 2965/tcp, 2129/tcp, 2208/tcp, 2280/tcp, 2486/tcp, 2046/tcp, 2078/tcp, 2575/tcp, 2135/tcp, 2232/tcp, 2276/tcp, 2430/tcp, 2474/tcp, 2971/tcp, 2672/tcp, 2737/tcp, 2773/tcp, 2333/tcp, 2377/tcp, 2636/tcp, 2018/tcp, 2797/tcp, 2115/tcp, 2151/tcp, 2216/tcp, 2090/tcp, 2414/tcp, 2450/tcp, 2599/tcp, 2612/tcp, 2656/tcp, 2959/tcp, 2995/tcp, 2252/tcp, 2511/tcp, 2555/tcp, 2753/tcp, 2313/tcp, 2240/tcp, 2327/tcp, 2363/tcp, 2402/tcp, 2006/tcp, 2125/tcp, 2161/tcp, 2204/tcp, 2688/tcp, 2723/tcp, 2767/tcp, 2521/tcp, 2969/tcp, 2565/tcp, 2600/tcp, 2644/tcp, 2931/tcp, 2579/tcp, 2139/tcp, 2074/tcp, 2478/tcp, 2030/tcp, 2462/tcp, 2105/tcp, 2141/tcp, 2307/tcp, 2343/tcp, 2660/tcp, 2220/tcp, 2426/tcp, 2747/tcp, 2949/tcp, 2985/tcp, 2545/tcp, 2501/tcp, 2589/tcp, 2703/tcp, 2498/tcp, 2058/tcp, 2793/tcp, 2111/tcp, 2155/tcp, 2094/tcp, 2410/tcp, 2454/tcp, 2515/tcp, 2551/tcp, 2616/tcp, 2652/tcp, 2919/tcp, 2199/tcp, 2212/tcp, 2256/tcp, 2991/tcp, 2757/tcp, 2317/tcp, 2244/tcp, 2288/tcp, 2323/tcp, 2367/tcp, 2002/tcp, 2121/tcp, 2165/tcp, 2200/tcp, 2640/tcp, 2727/tcp, 2763/tcp, 2406/tcp, 2929/tcp, 2525/tcp, 2561/tcp, 2604/tcp, 2081/tcp, 2441/tcp, 2889/tcp, 2717/tcp, 2357/tcp, 2818/tcp, 2890/tcp, 2446/tcp, 2086/tcp, 2264/tcp, 2624/tcp, 2858/tcp, 2353/tcp, 2713/tcp, 2894/tcp, 2442/tcp, 2082/tcp, 2870/tcp, 2179/tcp, 2539/tcp, 2834/tcp, 2585/tcp, 2109/tcp, 2787/tcp, 2901/tcp, 2945/tcp, 2181/tcp, 2383/tcp, 2549/tcp, 2989/tcp, 2062/tcp, 2026/tcp, 2810/tcp, 2854/tcp, 2898/tcp, 2842/tcp, 2886/tcp, 2175/tcp, 2838/tcp, 2535/tcp, 2866/tcp, 2822/tcp, 2814/tcp, 2850/tcp, 2846/tcp, 2882/tcp, 2145/tcp, 2189/tcp, 2303/tcp, 2347/tcp, 2541/tcp, 2422/tcp, 2466/tcp, 2101/tcp, 2981/tcp, 2505/tcp, 2664/tcp, 2224/tcp, 2707/tcp, 2743/tcp, 2909/tcp, 2806/tcp, 2802/tcp, 2260/tcp, 2620/tcp, 2628/tcp, 2268/tcp, 2826/tcp, 2862/tcp, 2140/tcp, 2306/tcp, 2342/tcp, 2500/tcp, 2588/tcp, 2427/tcp, 2463/tcp, 2104/tcp, 2984/tcp, 2544/tcp, 2661/tcp, 2221/tcp, 2702/tcp, 2746/tcp, 2948/tcp, 2800/tcp, 2584/tcp, 2108/tcp, 2786/tcp, 2900/tcp, 2944/tcp, 2180/tcp, 2382/tcp, 2548/tcp, 2988/tcp, 2063/tcp, 2027/tcp, 2528/tcp, 2609/tcp, 2681/tcp, 2924/tcp, 2960/tcp, 2168/tcp, 2249/tcp, 2285/tcp, 2483/tcp, 2043/tcp, 2279/tcp, 2077/tcp, 2033/tcp, 2932/tcp, 2378/tcp, 2639/tcp, 2738/tcp, 2830/tcp, 2874/tcp, 2592/tcp, 2691/tcp, 2916/tcp, 2952/tcp, 2619/tcp, 2196/tcp, 2259/tcp, 2295/tcp, 2497/tcp, 2758/tcp, 2318/tcp, 2057/tcp, 2390/tcp, 2013/tcp, 2568/tcp, 2649/tcp, 2685/tcp, 2920/tcp, 2964/tcp, 2128/tcp, 2209/tcp, 2281/tcp, 2487/tcp, 2047/tcp, 2239/tcp, 2073/tcp, 2037/tcp, 2936/tcp, 2338/tcp, 2679/tcp, 2778/tcp, 2225/tcp, 2100/tcp, 2144/tcp, 2188/tcp, 2302/tcp, 2423/tcp, 2467/tcp, 2665/tcp, 2742/tcp, 2908/tcp, 2980/tcp, 2504/tcp, 2346/tcp, 2540/tcp, 2706/tcp, 2241/tcp, 2326/tcp, 2362/tcp, 2403/tcp, 2007/tcp, 2124/tcp, 2160/tcp, 2205/tcp, 2689/tcp, 2722/tcp, 2766/tcp, 2520/tcp, 2968/tcp, 2564/tcp, 2601/tcp, 2645/tcp, 2235/tcp, 2271/tcp, 2334/tcp, 2370/tcp, 2675/tcp, 2437/tcp, 2473/tcp, 2976/tcp, 2572/tcp, 2132/tcp, 2730/tcp, 2774/tcp, 2631/tcp, 2580/tcp, 2782/tcp, 2148/tcp, 2904/tcp, 2940/tcp, 2184/tcp, 2386/tcp, 2508/tcp, 2229/tcp, 2067/tcp, 2023/tcp, 2669/tcp, 2417/tcp, 2453/tcp, 2093/tcp, 2794/tcp, 2116/tcp, 2750/tcp, 2310/tcp, 2398/tcp, 2512/tcp, 2556/tcp, 2611/tcp, 2655/tcp, 2699/tcp, 2152/tcp, 2215/tcp, 2251/tcp, 2996/tcp, 2245/tcp, 2289/tcp, 2322/tcp, 2366/tcp, 2003/tcp, 2120/tcp, 2164/tcp, 2201/tcp, 2641/tcp, 2726/tcp, 2762/tcp, 2407/tcp, 2928/tcp, 2524/tcp, 2560/tcp, 2605/tcp, 2231/tcp, 2275/tcp, 2330/tcp, 2374/tcp, 2671/tcp, 2433/tcp, 2477/tcp, 2972/tcp, 2576/tcp, 2136/tcp, 2734/tcp, 2770/tcp, 2635/tcp, 2171/tcp, 2878/tcp, 2531/tcp, 2447/tcp, 2087/tcp, 2176/tcp, 2536/tcp, 2354/tcp, 2714/tcp, 2893/tcp, 2443/tcp, 2083/tcp, 2172/tcp, 2532/tcp, 2629/tcp, 2269/tcp, 2827/tcp, 2863/tcp, 2596/tcp, 2695/tcp, 2912/tcp, 2956/tcp, 2659/tcp, 2192/tcp, 2219/tcp, 2291/tcp, 2798/tcp, 2394/tcp, 2053/tcp, 2017/tcp, 2493/tcp, 2843/tcp, 2887/tcp, 2833/tcp, 2877/tcp, 2265/tcp, 2625/tcp, 2813/tcp, 2857/tcp, 2847/tcp, 2883/tcp, 2837/tcp, 2873/tcp, 2790/tcp, 2112/tcp, 2156/tcp, 2211/tcp, 2255/tcp, 2413/tcp, 2457/tcp, 2097/tcp, 2615/tcp, 2651/tcp, 2992/tcp, 2754/tcp, 2314/tcp, 2299/tcp, 2516/tcp, 2552/tcp, 2807/tcp, 2803/tcp, 2350/tcp, 2710/tcp, 2897/tcp, 2853/tcp, 2358/tcp, 2718/tcp, 2817/tcp, 2250/tcp, 2513/tcp, 2557/tcp, 2610/tcp, 2654/tcp, 2117/tcp, 2153/tcp, 2214/tcp, 2311/tcp, 2399/tcp, 2416/tcp, 2452/tcp, 2092/tcp, 2698/tcp, 2997/tcp, 2751/tcp, 2795/tcp, 2597/tcp, 2694/tcp, 2913/tcp, 2957/tcp, 2658/tcp, 2193/tcp, 2218/tcp, 2290/tcp, 2799/tcp, 2395/tcp, 2052/tcp, 2016/tcp, 2492/tcp, 2278/tcp, 2076/tcp, 2032/tcp, 2933/tcp, 3000/tcp, 2379/tcp, 2638/tcp, 2739/tcp, 2583/tcp, 2709/tcp, 2907/tcp, 2943/tcp, 2781/tcp, 2187/tcp, 2349/tcp, 2385/tcp, 2064/tcp, 2468/tcp, 2020/tcp, 2867/tcp, 2823/tcp, 2682/tcp, 2729/tcp, 2927/tcp, 2963/tcp, 2480/tcp, 2286/tcp, 2369/tcp, 2408/tcp, 2040/tcp, 2238/tcp, 2072/tcp, 2036/tcp, 2937/tcp, 2339/tcp, 2678/tcp, 2779/tcp, 2587/tcp, 2749/tcp, 2903/tcp, 2947/tcp, 2785/tcp, 2183/tcp, 2309/tcp, 2381/tcp, 2060/tcp, 2024/tcp, 2428/tcp, 2791/tcp, 2755/tcp, 2315/tcp, 2113/tcp, 2157/tcp, 2412/tcp, 2456/tcp, 2096/tcp, 2517/tcp, 2553/tcp, 2614/tcp, 2650/tcp, 2993/tcp, 2210/tcp, 2254/tcp, 2298/tcp, 2234/tcp, 2270/tcp, 2335/tcp, 2371/tcp, 2674/tcp, 2436/tcp, 2472/tcp, 2977/tcp, 2573/tcp, 2133/tcp, 2731/tcp, 2775/tcp, 2630/tcp, 2107/tcp, 2143/tcp, 2305/tcp, 2341/tcp, 2547/tcp, 2424/tcp, 2460/tcp, 2028/tcp, 2987/tcp, 2503/tcp, 2662/tcp, 2222/tcp, 2701/tcp, 2745/tcp, 2789/tcp, 2496/tcp, 2759/tcp, 2319/tcp, 2197/tcp, 2258/tcp, 2056/tcp, 2391/tcp, 2012/tcp, 2917/tcp, 2953/tcp, 2618/tcp, 2294/tcp, 2593/tcp, 2690/tcp, 2242/tcp, 2325/tcp, 2361/tcp, 2400/tcp, 2004/tcp, 2127/tcp, 2163/tcp, 2206/tcp, 2721/tcp, 2765/tcp, 2523/tcp, 2488/tcp, 2048/tcp, 2567/tcp, 2602/tcp, 2646/tcp, 2230/tcp, 2274/tcp, 2331/tcp, 2375/tcp, 2670/tcp, 2432/tcp, 2476/tcp, 2973/tcp, 2577/tcp, 2137/tcp, 2735/tcp, 2771/tcp, 2634/tcp, 2103/tcp, 2147/tcp, 2301/tcp, 2345/tcp, 2389/tcp, 2420/tcp, 2464/tcp, 2068/tcp, 2741/tcp, 2983/tcp, 2666/tcp, 2226/tcp, 2507/tcp, 2543/tcp, 2705/tcp, 2261/tcp, 2621/tcp, 2177/tcp, 2537/tcp, 2266/tcp, 2626/tcp, 2828/tcp, 2444/tcp, 2084/tcp, 2848/tcp, 2173/tcp, 2533/tcp, 2262/tcp, 2622/tcp, 2868/tcp, 2852/tcp, 2359/tcp, 2719/tcp, 2816/tcp, 2769/tcp, 2008/tcp, 2923/tcp, 2967/tcp, 2484/tcp, 2282/tcp, 2329/tcp, 2686/tcp, 2044/tcp, 2832/tcp, 2876/tcp, 2820/tcp, 2864/tcp, 2355/tcp, 2715/tcp, 2892/tcp, 2840/tcp, 2884/tcp, 2836/tcp, 2872/tcp, 2824/tcp, 2860/tcp, 2246/tcp, 2321/tcp, 2365/tcp, 2000/tcp, 2527/tcp, 2123/tcp, 2167/tcp, 2202/tcp, 2725/tcp, 2761/tcp, 2404/tcp, 2563/tcp, 2606/tcp, 2642/tcp, 2804/tcp, 2080/tcp, 2440/tcp, 2888/tcp, 2880/tcp, 2088/tcp, 2448/tcp, 2844/tcp, 2243/tcp, 2324/tcp, 2360/tcp, 2401/tcp, 2005/tcp, 2126/tcp, 2162/tcp, 2207/tcp, 2720/tcp, 2764/tcp, 2522/tcp, 2489/tcp, 2049/tcp, 2566/tcp, 2603/tcp, 2647/tcp   modest_hodgkin      
```

The above output does not look too bad, however, if you try out the example, your terminal window will likely wrap the output and show you a big blob of port numbers!

With multiple containers using a number of moderately large port ranges, the problem is compounded and it becomes increasingly difficult to see the status of running containers.

A fix for this - I propose - would be to display the exact range as provided to the `--expose` flag of the `docker run` command, meaning that if I provide a range like `2000-3000` as above, the `docker ps` command displays `2000-3000/tcp` in the "Ports" column instead, e.g
```
vagrant@vagrant:~$ sudo docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS           NAMES
f17ad1f26b4d        ubuntu:14.04        "/bin/bash"         4 seconds ago       Up 3 seconds        2000-3000/tcp   modest_hodgkin      
```

Any thoughts on the matter? 
I will be investigating a solution this weekend, but I haven't delved into the docker codebase at all yet, so if someone else is interested in fixing this, I'll be glad to hear from you!

Cheers,
Colm

Debug information:
Running Docker v1.4.1 in Vagrant 1.6.2 with Ubuntu 14.04 and Virtualbox 4.3.8


```
vagrant@vagrant:~$ uname -a
Linux vagrant 3.13.0-24-generic #46-Ubuntu SMP Thu Apr 10 19:11:08 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

```
vagrant@vagrant:~$ docker version
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8
```

```
vagrant@vagrant:~$ docker -D info
Containers: 1
Images: 437
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Dirs: 439
Execution Driver: native-0.2
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04 LTS
CPUs: 2
Total Memory: 3.861 GiB
Name: vagrant
ID: 6N7F:7626:2WLC:BUQX:R5SY:VNU7:RBY6:WN65:JVIJ:A623:HW42:QOEB
Debug mode (server): false
Debug mode (client): true
Fds: 16
Goroutines: 14
EventsListeners: 0
Init Path: /usr/bin/docker
Docker Root Dir: /var/lib/docker
```

### PR DESCRIPTION
**Pull request by ColmHally**
Fixes #10302 

Signed-off-by: Colm Hally c.hallywebd@gmail.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks, @ColmHally

An example output was posted here: https://github.com/docker/docker/issues/10302#issuecomment-71396684

Should the output in https://docs.docker.com/reference/commandline/cli/#ps be updated to include a range (for illustration)? A short note below the example describing that port-ranges are shown "grouped" would be nice there as well.

**Comment by ColmHally** [comment]
Thanks, @thaJeztah 
Good catch re: docs - I'll add a short example now to the docs.

**Comment by thaJeztah** [comment]
I think there's no real need to add a "separate" output, but it can be put in the output that's already there.

**Comment by ColmHally** [comment]
@thaJeztah, I've added a short note to the docs and rebased my commits. Any further thoughts?

Also, I didn't realise that it was also possible to publish a range of ports to the host.. Do you think we need to add support for those ranges in the `ps` output?

**Comment by thaJeztah** [comment]
If possible, yes. I think the original issue also was about publishing a large range of ports (e.g 1000-2000)

**Comment by SvenDowideit** [comment]
Docs LGTM - @fredlf @jamtur01

**Comment by jamtur01** [comment]
Build failed.

File: docs/sources/reference/commandline/cli.md [hunk 1]
```diff
@@ -1434,12 +1434,14 @@ Running `docker ps` showing 2 linked containers.
...
     $ sudo docker ps
     CONTAINER ID        IMAGE                        COMMAND                CREATED              STATUS              PORTS               NAMES
-    4c01db0b339c        ubuntu:12.04                 bash                   17 seconds ago       Up 16 seconds                           webapp
+    4c01db0b339c        ubuntu:12.04                 bash                   17 seconds ago       Up 16 seconds       3300-3310/tcp       webapp
     d7886598dbe2        crosbymichael/redis:latest   /redis-server --dir    33 minutes ago       Up 33 minutes       6379/tcp            redis,webapp/db
 
 `docker ps` will show only running containers by default. To see all containers:
 `docker ps -a`
 
+`docker ps` will group exposed ports into a single range if possible, e.g. a container that exposes TCP ports `100, 101, 102` will display `100-102/tcp` in the `PORTS` column.
```
Comment (fredlf) [inline docs/sources/reference/commandline/cli.md]: Run-on. Need to break this into two sentences starting at E.g. Need a comma after the "g." as well, please!

**Comment by fredlf** [comment]
Docs LGTM after my copy edit. Thanks for the contribution.

**Comment by ColmHally** [comment]
Thanks @fredlf - will incorporate your edits and add support for published ports later this evening.

**Comment by ColmHally** [comment]
I've added support for grouping published ranges of ports and @fredlf, I've made the changes you pointed out as well.

Grouping will work for ports published like so:
```
$ docker run -p 1000-2000:1000-2000 ubuntu:14.40 bash
eefa3dbcdd1cba22540dcf6787d7d2c56d654dc95b693de7027d4f35ed889cc0
$ docker ps
eefa3dbcdd1c        ubuntu:14.04        "bash"              11 seconds ago      Up 5 seconds        0.0.0.0:1000-2000->1000-2000/tcp   modest_galileo 
```
where there is an exact match between the host and internal container port range. However, ports published without an exact match (such as `-p 1500-2500:1000-2000` or `-p :1500-2500`) are simply printed out individually (as before):

```
$ docker run -p 1500-2500:1000-2000 ubuntu:14.40 bash
5cebbad0bf0cf5a1ab482fd22b889b04f61851045ceabfd9a70d46dec0527365
$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS    NAMES
5cebbad0bf0c        ubuntu:14.04        "bash"              2 minutes ago       Up About a minute   0.0.0.0:1500->1000/tcp, 0.0.0.0:1501->1001/tcp, 0.0.0.0:1502->1002/tcp, 0.0.0.0:1503->1003/tcp, 0.0.0.0:1504->1004/tcp, 0.0.0.0:1505->1005/tcp, 0.0.0.0:1506->1006/tcp, 0.0.0.0:1507->1007/tcp, 0.0.0.0:1508->1008/tcp, 0.0.0.0:1509->1009/tcp, 0.0.0.0:1510->1010/tcp, 0.0.0.0:1511->1011/tcp, 0.0.0.0:1512->1012/tcp, 0.0.0.0:1513->1013/tcp, 0.0.0.0:1514->1014/tcp, 0.0.0.0:1515->1015/tcp, 0.0.0.0:1516->1016/tcp, 0.0.0.0:1517->1017/tcp, 0.0.0.0:1518->1018/tcp ...............
```

Another more comprehensive example:
```
$ docker run -dt -p 443:443 --expose=400-405/udp -p 4327-4655:4327-4655 -p 127.0.0.1:6000-6003:6000-6003 -p 127.0.0.1:32000-32005:32000-32005/udp ubuntu:14.04 bash
519be5a7927fb7ac40ca7550216c623f3737d5f216d58e36be7ff61d4718b92e
$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED              STATUS              PORTS                                                                                                                                             NAMES
519be5a7927f        ubuntu:14.04        "bash"              About a minute ago   Up 10 seconds       0.0.0.0:443->443/tcp, 400-405/udp, 0.0.0.0:4327-4655->4327-4655/tcp, 127.0.0.1:6000-6003->6000-6003/tcp, 127.0.0.1:32000-32005->32000-32005/udp   serene_torvalds     
```

**Comment by LK4D4** [comment]
@ColmHally I noticed that you have unrelated commit from @jfrazelle in your PR. Could you remove it pls?

**Comment by LK4D4** [comment]
@ColmHally Also, could you pls write test in `integration-cli`. Should be pretty simple.

File: api/common.go [hunk 2]
```diff
@@ -29,21 +29,78 @@ func ValidateHost(val string) (string, error) {
...
 	ports.Sort()
 	for _, port := range ports.Data {
-		if port.Get("IP") == "" {
-			result = append(result, fmt.Sprintf("%d/%s", port.GetInt("PrivatePort"), port.Get("Type")))
-		} else {
-			result = append(result, fmt.Sprintf("%s:%d->%d/%s", port.Get("IP"), port.GetInt("PublicPort"), port.GetInt("PrivatePort"), port.Get("Type")))
+		var (
+			current      = port.GetInt("PrivatePort")
+			portKey      = port.Get("Type")
+			startOfGroup int
```
Comment (LK4D4) [inline api/common.go]: pretty confusing naming, should `start/end` or `first/last`

**Comment by jfrazelle** [comment]
hey @ColmHally looks like this needs a rebase, feel free to ping me right after and I will review, sorry for the delay on this, we really appreciate this patch :)

**Comment by jfrazelle** [comment]
hey @ColmHally I ended up rebasing for you and adding a test in #10835 so you don't need to worry about back and forth with fixes now. Thanks so much for the patch.

**Comment by ColmHally** [comment]
Thanks @jfrazelle and @LK4D4! Have been meaning to get to this, but just didn't get a chance.

**Pull request closed without merging**